src/test/java/picard/sam/DownsampleSamTest.java

```diff
@@ -0,0 +1,158 @@
+
+    private File tempDir;
+    private File tempSamFile;
+
+    private SAMRecordSetBuilder setBuilder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    private SAMReadGroupRecord readGroupRecord = new SAMReadGroupRecord(readGroupId);
+
+    @Override
+    public String getCommandLineProgramName() {
+            return DownsampleSam.class.getSimpleName();
+    }
+
+    //create a samfile with one tile and randomly placed reads within it.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final int numReads = 10000;
+        final String flowCellBarcode = "TESTBARCODE";
+        final int maxX = 10000;
+        final int maxY = 20000;
+        final int minX = 1000;
+        final int minY = 2000;
+
+        final String separator = ":";
+        final int lane = 1;
+        final int tile = 2203;
+
+        //use fixed random seed to eliminate possibility of intermittent failures
+        final Random rg = new Random(31);
+        setBuilder.setReadGroup(readGroupRecord);
+        setBuilder.setUseNmFlag(true);
+
+        for (int i = 0; i < numReads; i++) {
+            final int x = rg.nextInt(maxX) + minX;
+            final int y = rg.nextInt(maxY) + minY;
+
+            final String readName = flowCellBarcode + separator + lane + separator + tile + separator + x + separator + y;
+            setBuilder.addPair(readName, 1, 1, 100);
+        }
+
+        tempDir = IOUtil.createTempDir("pds_test", "PositionalDownsampling");
+        tempSamFile = File.createTempFile("PositionalDownsampleSam", ".bam", tempDir);
+
+        BufferedLineReader bufferedLineReader = null;
+        try {
+            bufferedLineReader = new BufferedLineReader(new FileInputStream(dict));
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        final SAMTextHeaderCodec codec = new SAMTextHeaderCodec();
+        final SAMFileHeader header = codec.decode(bufferedLineReader, dict.toString());
+
+        readGroupRecord.setSample(sample);
+        readGroupRecord.setPlatform(platform);
+        readGroupRecord.setLibrary(library);
+
+        header.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+        header.addReadGroup(readGroupRecord);
+
+        setBuilder.setHeader(header);
+
+        final SAMFileWriter writer = new SAMFileWriterFactory()
+                .setCreateIndex(true).makeBAMWriter(header, false, tempSamFile);
+
+        for (final SAMRecord record : setBuilder) {
+            writer.addAlignment(record);
+        }
+        writer.close();
+    }
+
+    @AfterTest
+    private void tearDown() {
+        IOUtil.deleteDirectoryTree(tempDir);
+    }
+
+    @DataProvider(name = "ValidArgumentsTestProvider")
+    public Object[][] ValidArgumentsTestProvider() {
+        final List<Object[]> objects = new ArrayList<>();
+        for(final Strategy strategy : Strategy.values())
+            for( final Integer seed : new Integer[]{1, null})
```

Comment:
Should this new seed test be added to `PositionalDownsampleSam` as well?

Reply:
no...as PositionalDownsampleSam doesn't use a Random seed...

Reply:
huh, look at that :)